src/glib/mine/svm.h

```diff
@@ -13,376 +13,640 @@
-
-    // compute normal vector from support vectors
-    TFltV WgtV(Dim);
-    TFlt Bias = -svm_model->rho[0]; // LIBSVM does w*x-b, while we do w*x+b; thus the sign flip
-    EAssertR(TLinAlg::Norm(WgtV) == 0.0, "Expected a zero weight vector.");
-    for (int Idx = 0; Idx < svm_model->l; ++Idx) {
-        svm_node_t* SV = svm_model->SV[Idx];
-        while (SV->index != -1) {
-            WgtV[SV->index - 1] += svm_model->sv_coef[0][Idx] * SV->value;
-            ++SV;
+        svm_model_t* svm_model = svm_train(&svm_problem, &svm_parameter, DebugNotify, ErrorNotify);
+        
+        TFltVV SVs(svm_model->l, DimN);
+        TFltVV Coef(svm_model->nr_class - 1, svm_model->l);
+        TFltV Rho(svm_model->nr_class * (svm_model->nr_class - 1)/2);
+        TFltV ProbA(svm_model->nr_class * (svm_model->nr_class - 1)/2);
+        TFltV ProbB(svm_model->nr_class * (svm_model->nr_class - 1)/2);
+        TIntV SVIndices(svm_model->l);
+        TIntV Label(svm_model->nr_class);
+        TIntV nSV(svm_model->nr_class);        
+        TFltV WgtV(DimN);
+        TFlt Bias = -svm_model->rho[0]; // LIBSVM does w*x-b, while we do w*x+b; thus the sign flip
+        EAssertR(TLinAlg::Norm(WgtV) == 0.0, "Expected a zero weight vector.");
+        for (int Idx = 0; Idx < svm_model->l; ++Idx) {
+            svm_node_t* SV = svm_model->SV[Idx];
+            while (SV->index != -1) {
+                SVs.PutXY(Idx, SV->index - 1, SV->value);
+                WgtV[SV->index - 1] += svm_model->sv_coef[0][Idx] * SV->value;
+                ++SV;
+            }
+            SVIndices.SetVal(Idx, svm_model->sv_indices[Idx]);
+            for (int cIdx = 0; cIdx < svm_model->nr_class - 1; cIdx++){
+              Coef.PutXY(cIdx, Idx, svm_model->sv_coef[cIdx][Idx]);
+            }
+        }        
+        for (int Idx = 0; Idx < svm_model->nr_class * (svm_model->nr_class - 1)/2; Idx++){
+          Rho.SetVal(Idx, svm_model->rho[Idx]);
+        }        
+        for (int Idx = 0; Idx < svm_model->nr_class; Idx++){
+          Label.SetVal(Idx, svm_model->label[Idx]);
+          nSV.SetVal(Idx, svm_model->nSV[Idx]);
         }
+        
+        this->WgtV = WgtV;
+        this->Bias = Bias;
+        this->NClass = svm_model->nr_class;
+        this->NSV = svm_model->l;
+        this->SV = SVs;
+        this->Coef = Coef;
+        this->Rho = Rho;
+        this->ProbA = ProbA;
+        this->ProbB = ProbB;
+        this->SVIndices = SVIndices;
+        this->Label = Label;
+        this->nSV = nSV;
+        this->Free = svm_model->free_sv;
+
+        svm_free_and_destroy_model(&svm_model);
+        svm_destroy_param(&svm_parameter);
+        free(svm_problem.y);
+        free(svm_problem.x);
+        free(x_space);
     }
 
-    svm_free_and_destroy_model(&svm_model);
-    svm_destroy_param(&svm_parameter);
-    free(svm_problem.y);
-    free(svm_problem.x);
-    free(x_space);
-    free(svm_model);
+    // LIBSVM for Eps-Support Vector Regression for sparse input
```

Comment:
Same comments to this function as to the two Classify functions above. Can we merge them all together into one Fit function?